### PR TITLE
added functionality to copy files to and from hdfs

### PR DESCRIPTION
**Pull request by kwhitehall**
- added functionality in FileUtils
- moved checkHDFS functionality from MCSUtils to FileUtils
- updated writeNodeToNetcdf to allow copying to HDFS

### CODE REVIEW

File: src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala [hunk 1]
```diff
@@ -49,40 +50,44 @@ object MCSUtils {
...
-      directory: String,
+      localDir: String,
       fileName: String,
       varData: ucar.ma2.ArrayInt.D2,
       lats: Array[Double],
-      lons: Array[Double]): Unit = {
-    val filepath = directory + fileName
+      lons: Array[Double],
+      hdfsDir: String): Unit = {
+    val currFile = localDir + System.getProperty("file.separator") + fileName
```
Comment (rahulpalamuttam) [inline src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala:81]: Very nice use of separator!

File: src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala [hunk 2]
```diff
@@ -49,40 +50,44 @@ object MCSUtils {
...
     try {
-      val fsplit = filepath.split("_")
+      val fsplit = currFile.split("_")
```
Comment (rahulpalamuttam) [inline src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala:83]: @kwhitehall 
I wonder if we can just parse string using regexp pattern matching.
I've used this before in the repartitionBySpace implementation.
Take a look here for reference :
https://www.safaribooksonline.com/library/view/scala-cookbook/9781449340292/ch01s09.html

File: src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala [hunk 3]
```diff
@@ -118,11 +123,13 @@ object MCSUtils {
       datafile.close()
     }
     catch {
-      case _: Throwable => logger.info("Error generating netCDF file for " + filepath + "\n")
+      case _: Throwable => logger.info("Error generating netCDF file for " + currFile + "\n")
```
Comment (rahulpalamuttam) [inline src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala:126]: If i remember correctly the try catch is only needed for the datafile.close() statement.
If that's the case can we limit the try block to just datafile.close or just the lines of code that throw the error?

I don't think we need to capture the lat/lon variable initialization step in the try block.

File: src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala (hunk 1 above)
Comment (kwhitehall) [reply src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala:81]: Thanks @rahulpalamuttam. Learnt it from @sujen1412  ;)

File: src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala (hunk 2 above)
Comment (kwhitehall) [reply src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala:83]: Using regex is overkill considering that the filename here was created by the program in the format F#CE#_latmin_latmax_lonmin_lonmax.nc so there is no other character needed for this parsing.

File: src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala (hunk 3 above)
Comment (kwhitehall) [reply src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala:126]: The error can be thrown for:
slicing the data or creating the file writer (https://github.com/SciSpark/SciSpark/blob/master/src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala#L83:L85)
or writing the actual file 
https://github.com/SciSpark/SciSpark/blob/master/src/main/scala/org/dia/algorithms/mcs/MCSUtils.scala#L113:L118
refactor the try block to accommodate these in another PR.

**Comment by rahulpalamuttam** [comment]
Given your responses let's move on with this PR.
The refactoring can happen in a later PR. 
+1

**Comment by kwhitehall** [comment]
Ack'ed.

**Pull request merged**